### ISSUE TITLE
Redux Integration

### ISSUE THREAD

**Issue by sbycrosz** [aksonov/react-native-router-flux#220]
TLDR; `Redux` state should reflects on the current state of the UI

The `Redux Integration` in the Readme only mentions about **adding router callback** into the `Redux Reducer`, adding a copy of current route (which won't get used for deciding which 'Route' to render), And **Router has it's own `Store`** (which'll get used for deciding which 'Route' to render) separate from the `Redux Store`. This **breaks Redux's single data tree / source of truth** since there are 2 separate store.

Use case for this is when I'm recording my `Redux Actions`, sending them as Error Logs, and replaying the `Actions`. (Note that when I'm replaying the `Router AFTER_ROUTE etc.` as `Redux Action`s, the UI will not change since the **Router depends on its internal store** instead of Redux's store)

One possible solution is create a `RouterReducer`, let the user add it to their root reducer, and use the data from `Redux Store` to render the UI (As they did for React Router https://github.com/reactjs/react-router-redux)

Anyway this's just a conversation placeholder, I will try to conjure some examples over the weekend.
Cheers,

**Comment by aksonov** [comment]
You are right, you may add own 'plugin' instead of ExRouter, ReduxRouter that will use some redux state to manage UI. Current BaseRouter/Route could easily be converted to be fully immutable, so you could use it in your state. As I know @MITDelian worked on such approach, but i don't know when he plans to finish.

Anyway, PR is welcome.

**Comment by aksonov** [comment]
Also you may want to check #186

**Comment by sbycrosz** [comment]
I guess this'll continue on that thread :]

**Comment by MITDelian** [comment]
Hey @sbycrosz definitely something I am working on. I'm hoping to have an initial version out in the next two weeks that just acts as a separate plugin that is backwards compatible with this library by making the Router class immutable and storing it in the redux store and making a reducer as you mentioned above

**Comment by braco** [comment]
@MITDelian - you still working on this?

**Comment by MITDelian** [comment]
@braco  I have something that works for me that I use on an internal project. It's not super easy to rip out so I've stalled on this project until I can find some time to contribute it back.

So no, I'm not actively working on it but once I get both iOS and Android out on the app store I can spend some time on it again. Let's continue the conversation on #186 though.

**Comment by SudoPlz** [comment]
Am I wrong to think that v3 of react-native-router-flux still uses its own store?

**Comment by aksonov** [comment]
It uses React Native ExperimentalNavigation built-in store. According to https://github.com/ericvicenti/navigation-rfc/issues/52 author doesn't want to mix redux with this store...

**Comment by SudoPlz** [comment]
I see, thanks for the clarification man! Your library is awesome.

**Comment by SudoPlz** [comment]
Is there any way I can update which schene the user sees whenever my (redux) store changes ?

**Comment by BerndWessels** [comment]
Any updates on this?
